kbase-extension/static/kbase/js/widgets/narrative_core/publicDataSources/search2DataSource.js

```diff
@@ -8,61 +8,131 @@ a search against these cached results
 define([
     'bluebird',
     'handlebars',
-    'kb_common/utils',
-    'kb_common/jsonRpc/dynamicServiceClient',
-    'common/kbaseSearchEngine',
+    'common/searchAPI2',
     './common'
 ], function (
     Promise,
     Handlebars,
-    Utils,
-    DynamicServiceClient,
-    KBaseSearchEngine,
+    SearchAPI2,
     common
 ) {
     'use strict';
+    class RefDataSearch {
+        constructor({url, token, timeout}) {
+            this.url = url;
+            this.token = token;
+            this.searchAPI2 = new SearchAPI2({url: this.url, token: this.token});
+            this.timeout = timeout;
+        }
 
-    function parseGenomeIndexV1 (item) {
-        // oh, lordy, people messing with the metadata...
-        return {
-            genome_id: item.kbase_id,
-            genome_source: item.data.source,
-            genome_source_id: item.data.source_id,
-            scientific_name: item.data.scientific_name,
-            domain: item.data.domain,
-            num_contigs: item.data.num_contigs,
-            num_cds: item.data.cds_count,
-            num_features: item.data.feature_counts && item.data.feature_counts.gene,
-            ws_ref: item.kbase_id,
-            workspace_name: null,
-            taxonomy: item.data.taxonomy,
-            object_name: item.object_name
-        };
-    }
-
-    function parseGenomeSearchResultItem (item) {
-        // get the type and version
-        var indexType = item.type.toLowerCase();
-        var indexTypeVersion = item.type_ver;
+        referenceGenomeTotal({source}) {
+            const params = {
+                query: {
+                    bool: {
+                        must: [
+                            {
+                                term: {
+                                    tags: 'refdata'
+                                }
+                            },
+                            {
+                                term: {
+                                    source: source
+                                }
+                            },
+                        ]
+                    }
+                },
+                only_public: true,
+                indexes: ['genome'],
+                size: 0,
+                from: 0,
+                track_total_hits: true
+            };
+            return this.searchAPI2.search_objects({params, timeout: this.timeout})
+                .then((result) => {
+                    return result.count;
+                });
+        }
 
-        if (indexType !== 'genome') {
-            throw new Error('Item is not a genome: ' + indexType);
+        referenceGenomeDataSearch({source, query, offset, limit}) {
```

Comment:
Yes, that is a long-open issue in search -- only whole word matches are supported. 
It is also an issue in the new Dashboard, er, Narratives tool.
Personally, I think it is very important, and quite unexpected to behavior otherwise.
I don't think there will be any changes in the Narrative required to enable or accommodate that behavior when this is finally fixed. It will just ... work :)

More relevant here ... perhaps there should be some sort of local help or tooltip to explain the search behavior, or a link to a help page for the tool? I'm afraid a user might give up or at least be frustrated after a misspelling, partial term, or common form like "E. coli", and, finding nothing, assume we it is missing.

Maybe a help message when a search turns up nothing? A refseq genome search for someone who knows what they are doing should not usually come up empty.